### PR TITLE
massive rename of headers from .h to .hpp

### PR DESCRIPTION
**Pull request by nicolasaunai**

### CODE REVIEW

File: tests/simulator/test_diagnostic_timestamps.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from pyphare.pharein import ElectronModel
 from pyphare.pharein import ElectromagDiagnostics, FluidDiagnostics, ParticleDiagnostics
 from pyphare.simulator.simulator import Simulator, startMPI
-from pyphare.pharesee.hierarchy import hierarchy_from, h5_filename_from, h5_time_grp_key
+from pyphare.pharesee.hppierarchy import hierarchy_from, h5_filename_from, h5_time_grp_key
```
Comment (PhilipDeegan) [inline tests/simulator/test_diagnostic_timestamps.py:11]: :eyes:

Comment (nicolasaunai) [reply tests/simulator/test_diagnostic_timestamps.py]: fixed and other occurences

File: src/amr/data/particles/refine/split.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#ifndef PHARE_SPLIT_H
```
Comment (PhilipDeegan) [inline src/amr/data/particles/refine/split.hpp:1]: do we care that this isn't "PHARE_SPLIT_HPP" ?

File: src/amr/data/field/field_variable.hpp [hunk 3]
```diff
@@ -5,9 +5,9 @@
 
 #include <utility>
 
-//#include "core/data/grid/gridlayout.h"
-//  #include "core/data/grid/gridlayout_impl.h"
-#include "field_data_factory.h"
+//#include "core/data/grid/gridlayout.hpp"
+//  #include "core/data/grid/gridlayout_impl.hpp"
```
Comment (PhilipDeegan) [inline src/amr/data/field/field_variable.hpp:9]: rm?

File: src/amr/data/particles/refine/split.hpp (hunk 2 above)
Comment (nicolasaunai) [reply src/amr/data/particles/refine/split.hpp:1]: do we?

Comment (PhilipDeegan) [reply src/amr/data/particles/refine/split.hpp:1]: some tools complain about these things

:man_shrugging:

Comment (nicolasaunai) [reply src/amr/data/particles/refine/split.hpp:1]: do you have these "some" in mind? I've never seen a tool that cares about the header guard

Comment (PhilipDeegan) [reply src/amr/data/particles/refine/split.hpp:1]: cpplint is such a tool, but we don't use it atm

```
Done processing src/python3/cpp_simulator.h
src/python3/data_wrangler.h:0:  No copyright message found.  You should have a line: "Copyright [year] <Copyright Owner>"  [legal/copyright] [5]
src/python3/data_wrangler.h:1:  #ifndef header guard has wrong style, please use: SRC_PYTHON3_DATA_WRANGLER_H_  [build/header_guard] [5]
```

Comment (nicolasaunai) [reply src/amr/data/particles/refine/split.hpp]: done

**Pull request merged**